hs_core/models.py

```diff
@@ -594,6 +594,15 @@ def create(cls, **kwargs):
             # ensure isHostedBy and isCopiedFrom are mutually exclusive
             metadata_obj = kwargs['content_object']
             metadata_type = ContentType.objects.get_for_model(metadata_obj)
+
+            # avoid creating duplicated element
+            if Relation.objects.filter(type=kwargs['type'],
+                                       value=kwargs['value'],
+                                       object_id=metadata_obj.id,
+                                       content_type=metadata_type).exists():
+                raise ValidationError('Relation element of the same type and value already exists. '
+                                      'Cannot create a duplicated element')
```

Comment:
Validation error message "Relation element of the same type and value already exists."  should be sufficient.